app/models/host/managed.rb

```diff
@@ -192,7 +197,7 @@ class Jail < ::Safemode::Jail
                           :if => Proc.new { |host| host.managed && host.pxe_build? && build? }
     validates :ptable_id, :presence => {:message => N_("can't be blank unless a custom partition has been defined")},
                           :if => Proc.new { |host| host.managed and host.disk.empty? and not Foreman.in_rake? and host.pxe_build? and host.build? }
-    validates :provision_method, :inclusion => {:in => PROVISION_METHODS, :message => N_('is unknown')}, :if => Proc.new {|host| host.managed?}
+    validates :provision_method, :inclusion => {:in => proc { self.provision_methods }, :message => N_('is unknown')}, :if => Proc.new {|host| host.managed?}
```

Comment:
This could do with a unit test to check validation of a host works with a newly registered provisioning method.